lib/Dialect/AIE/Transforms/AIECreatePathFindFlows.cpp

```diff
@@ -426,16 +445,87 @@ void AIEPathfinderPass::runOnPacketFlow(DeviceOp device, OpBuilder &builder) {
         return -1;
       };
 
-  std::vector<std::pair<std::pair<PhysPort, int>, SmallVector<PhysPort, 4>>>
-      sortedPacketFlows(packetFlows.begin(), packetFlows.end());
-
-  // To get determinsitic behaviour
-  std::sort(sortedPacketFlows.begin(), sortedPacketFlows.end(),
-            [](const auto &lhs, const auto &rhs) {
-              auto lhsFlowID = lhs.first.second;
-              auto rhsFlowID = rhs.first.second;
-              return lhsFlowID < rhsFlowID;
+  // To get determinsitic behaviour; allocate amsels for control packet flows
+  // before others
+  auto getWireBundleAsInt = [](WireBundle bundle) {
+    switch (bundle) {
+    case WireBundle::Core:
+      return 0;
+    case WireBundle::DMA:
+      return 1;
+    case WireBundle::FIFO:
+      return 2;
+    case WireBundle::South:
+      return 3;
+    case WireBundle::West:
+      return 4;
+    case WireBundle::North:
+      return 5;
+    case WireBundle::East:
+      return 6;
+    case WireBundle::PLIO:
+      return 7;
+    case WireBundle::NOC:
+      return 8;
+    case WireBundle::Trace:
+      return 9;
+    case WireBundle::Ctrl:
+      return 10;
+    }
+    return -1;
+  };
+  auto getUniqueIdPerFlowPerSB =
+      [getWireBundleAsInt](int flowID, WireBundle srcBundle,
+                           SmallVector<PhysPort, 4> dests) {
+        int currMultiplier =
+            16; // TODO: Will fail when WireBundle enum eventually exceeds 16
```

Comment:
Try `AIE::getMaxEnumValForWireBundle()` ?

Reply:
Ah really nice to have that method. PR updated.